src/gpg/eopen.py

```diff
@@ -0,0 +1,72 @@
+import gpg
+import os
+import sys
+import time
+import textwrap
+
+DEBUG= True if (os.environ['DEBUG']=='yes') else False
+
+def print_signatures(verify_result):
+    c=gpg.Context()
+    for signature in verify_result.signatures:
+        user = c.get_key(signature.fpr).uids[0].uid
+        fpr = signature.fpr
+        signed_time = time.ctime(signature.timestamp)
+
+        message = '''
+                    Good signature from {user}
+                    with key {fingerprint}
+                    made at {time}
+                    '''.format(user=user,
+                                fingerprint=fpr,
+                                time=signed_time)
+
+        print(textwrap.dedent(message))
+
+def eopen(sealed_file_path, output_file_path):
+    """
+    sign and encrypt file_path to the recipents
+    """
+
+    try:
+        c = gpg.Context()
+        with open(sealed_file_path, "rb") as cfile:
+            try:
+                plaintext, result, verify_result = c.decrypt(cfile,verify=True)
+                
+                if(DEBUG):
+                    print(verify_result, result)
+                
+                print_signatures(verify_result)
+
+            except gpg.errors.BadSignatures:
+                print("Could not verify signatures")
+                exit(3)
+                
+        while(os.path.exists(output_file_path)):
+            print("File {output} already exists".format(output=output_file_path))
```

Comment:
Making this check in the bash script is simpler. Just fail with a proper message if output file already exists.

Reply:
The gpg command that I replaced had this behavior, it asked to override or enter a new file name if the file already exists.
If I understood correctly you are suggesting to move this checking functionality to bash script before calling this python script??

Reply:
> If I understood correctly you are suggesting to move this checking functionality to bash script before calling this python script??

This is right.

> The gpg command that I replaced had this behavior, it asked to override or enter a new file name if the file already exists.

We don't have to duplicate the behavior of `gpg`.
In this case I think that it is more reasonable that we just inform the user that a file of that name already exists, and stop. The user can rename or delete the existing file and try the command again.

Reply:
Also don't forget to apply good pythonic style to this script too.

Reply:
Yup I'll add it just in a while.